_posts/2023-08-02-#28122.md

```diff
@@ -0,0 +1,62 @@
+---
+layout: pr
+date: 2023-08-02
+title: "Silent Payments: Implement BIP352"
+pr: 28122
+authors: [josibake]
+components: ["wallet"]
+host: josibake
+status: upcoming
+commit:
+---
+
+## Notes
+
+This PR is the first step towards implementing [BIP352: Silent Payments](https://github.com/bitcoin/bips/pull/1458) in the Bitcoin Core wallet. BIP352 is "a protocol for static payment addresses in Bitcoin without on-chain linkability of payments or a need for on-chain notifications."
+
+Before reviewing the PR, it is _strongly_ recommended that you read the [BIP](https://github.com/bitcoin/bips/pull/1458), as well as the [reference implementation](https://github.com/bitcoin/bips/blob/cfe0771a0408a2d2de278d4e95bb9a33bd1615b2/bip-0352/reference.py) and [test vectors](https://github.com/bitcoin/bips/blob/cfe0771a0408a2d2de278d4e95bb9a33bd1615b2/bip-0352/send_and_receive_test_vectors.json). This PR implements the logic from the BIP and is a child PR of [Silent Payments: send and receive](https://github.com/bitcoin/bitcoin/pull/27827).
+
+## Questions
+
+1. Did you read the [BIP](https://github.com/bitcoin/bips/pull/1458)? In a few sentences, can you summarize it in your own words?
+
+1. Did you review the [reference implementation](https://github.com/bitcoin/bips/blob/cfe0771a0408a2d2de278d4e95bb9a33bd1615b2/bip-0352/reference.py) and [test vectors](https://github.com/bitcoin/bips/blob/cfe0771a0408a2d2de278d4e95bb9a33bd1615b2/bip-0352/send_and_receive_test_vectors.json)? Can you think of additional test cases?
+
+1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach?
+
+1. In [5688262](https://github.com/bitcoin/bitcoin/pull/28122/commits/56882622faf469b6f948f79a69c3c8ddbde92ff8), why do we need a custom hash function? Why not just use the default hash function from `libsecp256k1`? (hint: [see this discussion on the BIP](https://github.com/bitcoin/bips/pull/1458#pullrequestreview-1466163601))
+
+1. [ae6019b](https://github.com/bitcoin/bitcoin/pull/28122/commits/ae6019b3a5f99e63b5f23e48fd430147b8500877) and [439e57a](https://github.com/bitcoin/bitcoin/pull/28122/commits/439e57aa88d33e9c61e4c38a37526133a79ab188) add functions for decoding and encoding silent payment addresses, but [src/key_io.cpp](https://github.com/bitcoin/bitcoin/blob/439e57aa88d33e9c61e4c38a37526133a79ab188/src/key_io.cpp) already has a `DestinationEncoder` class and a `DecodeDestination` function. Why can't we just add silent payment addresses to the existing encoder and decoder? More specifically, what is a `CTxDestination`? How is it different from a silent payment address? (hint: `CTxDestination` is defined in [src/script/standards.h](https://github.com/josibake/bitcoin/blob/439e57aa88d33e9c61e4c38a37526133a79ab188/src/script/standard.h))
```

Comment:
I had difficulty understanding what was being asked, probably best to split this up into multiple questions or add the background information as notes

Reply:
Added some notes to the background section and tried to rephrase this to be a little more clear